src/main/java/org/spongepowered/api/service/command/sponge/SpongeCommandService.java

```diff
@@ -0,0 +1,200 @@
+import org.spongepowered.api.plugin.PluginManager;
+import org.spongepowered.api.service.command.CommandService;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.Texts;
+import org.spongepowered.api.util.command.CommandCallable;
+import org.spongepowered.api.util.command.CommandException;
+import org.spongepowered.api.util.command.CommandMapping;
+import org.spongepowered.api.util.command.CommandSource;
+import org.spongepowered.api.util.event.Order;
+import org.spongepowered.api.util.event.Subscribe;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+public class SpongeCommandService implements CommandService {
+
+    private final SpongeDispatcher dispatcher;
+    private final PluginManager manager;
+
+    private static final Logger log = LoggerFactory.getLogger(SpongeCommandService.class);
+
+    public SpongeCommandService(PluginManager manager, boolean enableNotFoundMessages) {
+        this.dispatcher = new SpongeDispatcher(enableNotFoundMessages);
+        this.manager = manager;
+    }
+
+    /**
+     * Receive {@link CommandEvent}s.
+     *
+     * @param event The event received
+     */
+    @Subscribe(order = Order.LAST)
+    public void onCommandEvent(final CommandEvent event) {
+        try {
+            if (call(event.getSource(), event.getCommand() + " " + event.getArguments(), Collections.<String>emptyList())) {
+                event.setCancelled(true);
+            }
+        } catch (CommandException e) {
+            event.setCancelled(true);
+            log.warn("Failed to execute a command", e);
+        }
+    }
+
+    @Override
+    public Set<CommandMapping> getCommands() {
+        return this.dispatcher.getCommands();
+    }
+
+    @Override
+    public Set<String> getPrimaryAliases() {
+        return this.dispatcher.getPrimaryAliases();
+    }
+
+    @Override
+    public Set<String> getAliases() {
+        return this.dispatcher.getAliases();
+    }
+
+    @Override
+    public Set<? extends CommandMapping> getAll(String alias) {
+        return this.dispatcher.getAll(alias);
+    }
+
+    @Override
+    public Optional<? extends CommandMapping> resolveMapping(String alias, CommandSource source) {
+        return this.dispatcher.resolveMapping(alias, source);
+    }
+
+    @Override
+    public boolean containsAlias(String alias) {
+        return this.dispatcher.containsAlias(alias);
+    }
+
+    @Override
+    public boolean containsMapping(CommandMapping mapping) {
+        return this.dispatcher.containsMapping(mapping);
+    }
+
+    @Override
+    public boolean call(CommandSource source, String arguments, List<String> parents) throws CommandException {
+        return this.dispatcher.call(source, arguments, parents);
+    }
+
+    @Override
+    public boolean testPermission(CommandSource source) {
+        return true;
+    }
+
+    @Override
+    public String getShortDescription(CommandSource source) {
+        return "";
```

Comment:
These things should be specified in the constructor parameters ideally. Similar to [`SimpleDispatcher`](https://github.com/SpongePowered/SpongeAPI/blob/master/src/main/java/org/spongepowered/api/util/command/dispatcher/SimpleDispatcher.java#L77).